src/TLuaInterpreter.cpp

```diff
@@ -18441,3 +18455,33 @@ void TLuaInterpreter::updateExtendedAnsiColorsInTable()
         lua_pop(L, 1);
     }
 }
+
+// Internal function - Looks for headers in an http response. If it finds them,
+// it creates a table on the lua stack for the headers, pops the table from the
+// stack, puts it in the lua registry, and returns the key to where it is in
+// registery, wrapped in an optional. Otherwise it returns an empty optional.
+std::optional<int> TLuaInterpreter::createHttpHeadersTable(QNetworkReply* reply)
+{
+    lua_State* L = pGlobalLua;
+    if (!L) {
+        qDebug() << "LUA CRITICAL ERROR: no suitable Lua execution unit found.";
+        return {};
+    }
+    const QList<QByteArray> headerList = reply->rawHeaderList();
+    if (headerList.isEmpty()) {
+        return {};
+    }
+
+    // Push table onto stack
+    lua_newtable(L);
+    for (QByteArray header : headerList) {
+        // Push header key onto stack
+        lua_pushstring(L, header.toStdString().c_str());
```

Comment:
For what it is worth we tend to use `QString::toUtf8().constData()` in this situation...

Reply:
Updated :+1: